api/turing/service/router_deployment_service.go

```diff
@@ -677,3 +717,133 @@ const (
+) []*cluster.PodDisruptionBudget {
+	pdbs := []*cluster.PodDisruptionBudget{}
+
+	// Enricher's PDB
+	if routerVersion.Enricher != nil {
+		enricherPdb := ds.svcBuilder.NewPodDisruptionBudget(
+			routerVersion,
+			project,
+			servicebuilder.ComponentTypes.Enricher,
+			ds.pdbConfig,
+		)
+		pdbs = append(pdbs, enricherPdb)
+	}
+
+	// Ensembler's PDB
+	if routerVersion.Enricher != nil {
+		ensemblerPdb := ds.svcBuilder.NewPodDisruptionBudget(
+			routerVersion,
+			project,
+			servicebuilder.ComponentTypes.Ensembler,
+			ds.pdbConfig,
+		)
+		pdbs = append(pdbs, ensemblerPdb)
+	}
+
+	// Router's PDB
+	routerPdb := ds.svcBuilder.NewPodDisruptionBudget(
+		routerVersion,
+		project,
+		servicebuilder.ComponentTypes.Router,
+		ds.pdbConfig,
+	)
+	pdbs = append(pdbs, routerPdb)
+
+	return pdbs
+}
+
+func deployPodDisruptionBudgets(ctx context.Context,
+	controller cluster.Controller, pdbs []*cluster.PodDisruptionBudget, eventsCh *EventChannel,
+) error {
+	// Define deploy function
+	deployFunc := func(ctx context.Context,
+		obj any,
+		wg *sync.WaitGroup,
+		errCh chan<- error,
+		eventsCh *EventChannel,
+	) {
+		defer wg.Done()
+
+		pdb, ok := obj.(*cluster.PodDisruptionBudget)
+		if !ok {
+			err := errors.Errorf("failed to parse *cluster.PodDisruptionBudget config")
+			eventsCh.Write(models.NewErrorEvent(
+				models.EventStageDeployingServices, "failed to deploy pdb: %s", err.Error()))
+			errCh <- err
+			return
+		}
+
+		eventsCh.Write(models.NewInfoEvent(
+			models.EventStageDeployingServices, "deploying pdb %s", pdb.Name))
+
+		_, err := controller.CreatePodDisruptionBudget(ctx, pdb.Namespace, *pdb)
+		if err != nil {
+			err = errors.Wrapf(err, "Failed to deploy pdb %s", pdb.Name)
+			eventsCh.Write(models.NewErrorEvent(
+				models.EventStageDeployingServices, "failed to deploy pdb %s: %s", pdb.Name, err.Error()))
+		}
+
+		eventsCh.Write(models.NewInfoEvent(
+			models.EventStageDeployingServices, "successfully deployed pdb %s", pdb.Name))
+
+		errCh <- err
+	}
+
+	select {
+	case <-ctx.Done():
+		return errors.New("timeout deploying pdb")
+	default:
+		return updateResources(ctx, pdbs, deployFunc, eventsCh)
+	}
+}
+
+func deletePodDisruptionBudgets(ctx context.Context,
+	controller cluster.Controller, pdbs []*cluster.PodDisruptionBudget, eventsCh *EventChannel,
+) error {
+	// Define deploy function
+	deleteFunc := func(ctx context.Context,
+		obj any,
+		wg *sync.WaitGroup,
+		errCh chan<- error,
+		eventsCh *EventChannel,
+	) {
+		defer wg.Done()
+
+		pdb, ok := obj.(*cluster.PodDisruptionBudget)
+		if !ok {
+			err := errors.Errorf("failed to parse *cluster.PodDisruptionBudget config")
+			eventsCh.Write(models.NewErrorEvent(
+				models.EventStageDeployingServices, "failed to delete pdb: %s", err.Error()))
+			errCh <- err
+			return
+		}
+
+		eventsCh.Write(models.NewInfoEvent(
+			models.EventStageDeployingServices, "deleting pdb %s", pdb.Name))
+
+		err := controller.DeletePodDisruptionBudget(ctx, pdb.Namespace, pdb.Name)
```

Comment:
In previous releases, we wouldn't have deployed PDBs - so we cannot delete them either. Will issuing the delete call return an error? If so, we should ignore such error as it's a valid scenario.